File: classes/parte-6-datos-y-estructuras/100-enumeraciones-y-tipos-algebraicos-adt-sum-types/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	t := strings.Fields(line)
	var area int64
	if t[0] == "cuadrado" {
		l, _ := strconv.ParseInt(t[1], 10, 64)
		area = l * l
	} else {
		a, _ := strconv.ParseInt(t[1], 10, 64)
		b, _ := strconv.ParseInt(t[2], 10, 64)
		area = a * b
	}
	fmt.Printf("area=%d\n", area)
}

File: classes/parte-6-datos-y-estructuras/100-enumeraciones-y-tipos-algebraicos-adt-sum-types/implementaciones/c/main.c
#include <stdio.h>
#include <string.h>

int main(void) {
    char tipo[32];
    if (scanf("%31s", tipo) != 1) return 1;
    long area;
    if (strcmp(tipo, "cuadrado") == 0) {
        long l;
        if (scanf("%ld", &l) != 1) return 1;
        area = l * l;
    } else {
        long a, b;
        if (scanf("%ld %ld", &a, &b) != 2) return 1;
        area = a * b;
    }
    printf("area=%ld\n", area);
    return 0;
}
